### PR TITLE
add gravity gauntlet

### PR DESCRIPTION
**Pull request by repulica**
adds gravity gauntlet weapon to astromine
gauntlet adds 4 attack damage
hold right click with gauntlet on each hand to charge (costs 1024 energy)
attack with charged gauntlet on each hand knock back enemies
costs advanced circuit galaxium rose gold plates steel plates and advanced battery

### CODE REVIEW

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+	@Override
+	public boolean hasGlint(ItemStack stack) {
+		return stack.getOrCreateTag().getBoolean("Charged");
+	}
+
+	// TODO: dynamic once not broken
+	@Override
+	public Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers(EquipmentSlot slot) {
+		Multimap<EntityAttribute, EntityAttributeModifier> eams = HashMultimap.create();
```
Comment (shedaniel) [inline src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:133]: consi

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		if (slot == EquipmentSlot.MAINHAND) {
+			eams.put(EntityAttributes.GENERIC_ATTACK_DAMAGE, new EntityAttributeModifier(ATTACK_DAMAGE_MODIFIER_ID, "attack", 4f, EntityAttributeModifier.Operation.ADDITION));
```
Comment (shedaniel) [inline src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:135]: No, you aren't even returning the multimap, there is no point in using the dynamic ones, just create a map in the constructor and return the one in the vanilla attribute modifiers.

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		}
+		return super.getAttributeModifiers(slot);
+	}
+
+	@Environment(EnvType.CLIENT)
+	@Override
+	public void appendTooltip(ItemStack stack, World world, List<Text> tooltip, TooltipContext context) {
+		EnergyHandler energyHandler = Energy.of(stack);
+		tooltip.add(EnergyUtilities.compoundDisplayColored(energyHandler.getEnergy(), energyHandler.getMaxStored()));
```
Comment (shedaniel) [inline src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:144]: call super.appendTooltip

File: src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java [hunk 4]
```diff
@@ -60,6 +61,7 @@
 
 	// Fantasy weaponry
 	public static final Item SUPER_SPACE_SLIME_SHOOTER = register("super_space_slime_shooter", new SuperSpaceSlimeShooterItem(getBasicSettings()));
+	public static final GravityGauntletItem GRAVITY_GAUNTLET = register("gravity_gauntlet", new GravityGauntletItem(getBasicSettings().fireproof().maxCount(1), AstromineConfig.get().gravityGauntletEnergy));
```
Comment (shedaniel) [inline src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java:64]: I would move the damage() to the constructor of GravityGauntletItem like vanilla and what's the point of fireproof

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java (hunk 2 above)
Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:135]: oops

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java (hunk 3 above)
Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:144]: that would add the crafting use only warning

File: src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java (hunk 4 above)
Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java:64]: to match the other weapons

File: src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java (hunk 3 above)
Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/common/item/weapon/GravityGauntletItem.java:144]: https://github.com/Chainmail-Studios/Astromine/blob/development/src/main/java/com/github/chainmailstudios/astromine/common/item/base/EnergyVolumeItem.java#L70

File: src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java (hunk 4 above)
Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java:64]: also the gauntlet doesnt take damage

Comment (shedaniel) [reply src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java]: don't think we want fire proof on a guantlet

Comment (repulica) [reply src/main/java/com/github/chainmailstudios/astromine/registry/AstromineItems.java]: ok

**Comment by shedaniel** [comment]
Does this even work?

**Comment by repulica** [comment]
yes

i cant upload videos to github so it sent it to astromine discord

**Comment by repulica** [comment]
is there a timeframe for this to be merged

**Pull request merged**